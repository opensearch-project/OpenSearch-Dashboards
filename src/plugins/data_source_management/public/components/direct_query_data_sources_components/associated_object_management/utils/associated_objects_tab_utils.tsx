/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { ApplicationStart } from 'opensearch-dashboards/public';
import { DirectQueryLoadingStatus } from '../../../../../framework/types';
import {
  DATA_SOURCE_TYPES,
  DEFAULT_DATA_SOURCE_NAME,
  DEFAULT_DATA_SOURCE_TYPE,
} from '../../../../../framework/constants';
import { observabilityLogsID } from '../../../../../framework/utils/shared';

export const ASSC_OBJ_TABLE_SUBJ = 'associatedObjectsTable';

export const ASSC_OBJ_TABLE_ACC_COLUMN_NAME = 'accelerations';

export const ASSC_OBJ_TABLE_SEARCH_HINT = 'Search for objects';

export const ASSC_OBJ_PANEL_TITLE = 'Associated objects';

export const ASSC_OBJ_PANEL_DESCRIPTION = 'Manage objects associated with this data sources.';

export const ASSC_OBJ_NO_DATA_TITLE = 'You have no associated objects';

export const ASSC_OBJ_NO_DATA_DESCRIPTION =
  'Add or config tables from your data source or use Query Workbench.';

export const ASSC_OBJ_REFRESH_BTN = 'Refresh';

export const ASSC_OBJ_FRESH_MSG = 'Last updated at:';

export const ACCE_NO_DATA_TITLE = 'You have no accelerations';

export const ACCE_NO_DATA_DESCRIPTION = 'Accelerate query performing through OpenSearch Indexing';

export const CREATE_ACCELERATION_DESCRIPTION = 'Create Acceleration';

const catalogCacheFetchingStatus = [
  DirectQueryLoadingStatus.RUNNING,
  DirectQueryLoadingStatus.WAITING,
  DirectQueryLoadingStatus.SCHEDULED,
];

export const isCatalogCacheFetching = (...statuses: DirectQueryLoadingStatus[]) => {
  return statuses.some((status: DirectQueryLoadingStatus) =>
    catalogCacheFetchingStatus.includes(status)
  );
};

export const redirectToExplorerWithDataSrc = (
  datasourceName: string,
  datasourceType: string,
  databaseName: string,
  tableName: string,
  application: ApplicationStart
) => {
  const queryIndex = `${datasourceName}.${databaseName}.${tableName}`;
  redirectToExplorerWithQuery(datasourceName, datasourceType, queryIndex, application);
};

export const redirectToExplorerOSIdx = (indexName: string, application: ApplicationStart) => {
  redirectToExplorerWithQuery(
    DEFAULT_DATA_SOURCE_NAME,
    DEFAULT_DATA_SOURCE_TYPE,
    indexName,
    application
  );
};

export const redirectToExplorerS3 = (datasourceName: string, application: ApplicationStart) => {
  application.navigateToApp(observabilityLogsID, {
    path: `#/explorer`,
    state: {
      datasourceName,
      datasourceType: DATA_SOURCE_TYPES.S3Glue,
    },
  });
};

const redirectToExplorerWithQuery = (
  datasourceName: string,
  datasourceType: string,
  queriedIndex: string,
  application: ApplicationStart
) => {
  // navigate to explorer
  application.navigateToApp(observabilityLogsID, {
    path: `#/explorer`,
    state: {
      datasourceName,
      datasourceType,
      queryToRun: `source = ${queriedIndex} | head 10`,
    },
  });
};
