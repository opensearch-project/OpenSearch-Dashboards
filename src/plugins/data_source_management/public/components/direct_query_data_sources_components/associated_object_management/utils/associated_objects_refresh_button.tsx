/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton } from '@elastic/eui';
import React from 'react';
import { ASSC_OBJ_REFRESH_BTN } from './associated_objects_tab_utils';

interface AssociatedObjectsRefreshButtonProps {
  isLoading: boolean;
  onClick: () => void;
}

export const AssociatedObjectsRefreshButton: React.FC<AssociatedObjectsRefreshButtonProps> = (
  props
) => {
  const { isLoading, onClick } = props;

  return (
    <EuiButton iconType="refresh" onClick={onClick} isLoading={isLoading} isDisabled={isLoading}>
      {ASSC_OBJ_REFRESH_BTN}
    </EuiButton>
  );
};
