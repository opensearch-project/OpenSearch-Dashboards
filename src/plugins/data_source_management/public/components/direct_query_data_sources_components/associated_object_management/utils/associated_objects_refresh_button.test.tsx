/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import { AssociatedObjectsRefreshButton } from './associated_objects_refresh_button';
import { EuiButton } from '@elastic/eui';
import { ASSC_OBJ_REFRESH_BTN } from './associated_objects_tab_utils';

describe('AssociatedObjectsRefreshButton', () => {
  const defaultProps = {
    isLoading: false,
    onClick: jest.fn(),
  };

  const shallowComponent = (props = defaultProps) =>
    shallow(<AssociatedObjectsRefreshButton {...props} />);
  const mountComponent = (props = defaultProps) =>
    mount(<AssociatedObjectsRefreshButton {...props} />);

  test('renders correctly', () => {
    const wrapper = shallowComponent();
    expect(wrapper).toMatchSnapshot();
  });

  test('displays the correct button text', () => {
    const wrapper = mountComponent();
    expect(wrapper.find(EuiButton).text()).toEqual(ASSC_OBJ_REFRESH_BTN);
  });

  test('is disabled and shows loading state when isLoading is true', () => {
    const wrapper = mountComponent({ ...defaultProps, isLoading: true });
    const button = wrapper.find(EuiButton);
    expect(button.prop('isDisabled')).toBe(true);
    expect(button.prop('isLoading')).toBe(true);
  });

  test('is enabled and not loading when isLoading is false', () => {
    const wrapper = mountComponent();
    const button = wrapper.find(EuiButton);
    expect(button.prop('isDisabled')).toBe(false);
    expect(button.prop('isLoading')).toBe(false);
  });

  test('calls onClick when button is clicked', () => {
    const wrapper = mountComponent();
    wrapper.find(EuiButton).simulate('click');
    expect(defaultProps.onClick).toHaveBeenCalled();
  });
});
