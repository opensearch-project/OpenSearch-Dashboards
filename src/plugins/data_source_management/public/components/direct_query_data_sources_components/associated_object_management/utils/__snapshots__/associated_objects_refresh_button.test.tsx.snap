// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssociatedObjectsRefreshButton renders correctly 1`] = `
<EuiButton
  iconType="refresh"
  isDisabled={false}
  isLoading={false}
  onClick={[MockFunction]}
>
  Refresh
</EuiButton>
`;
